fog/ledger/server/src/db_fetcher.rs

```diff
@@ -164,62 +163,51 @@ impl<
     pub fn run(mut self) {
         log::info!(self.logger, "Db fetcher thread started.");
         let block_range = self.sharding_strategy.get_block_range();
-        let mut next_block_index = 0;
+        let mut next_block_index = block_range.start_block;
         loop {
-            if self.stop_requested.load(Ordering::SeqCst) {
-                log::info!(self.logger, "Db fetcher thread stop requested.");
-                break;
+            loop {
+                let num_blocks = self.load_block_data(&mut next_block_index);
+
+                let end = min(num_blocks, block_range.end_block);
+
+                if next_block_index < end.saturating_sub(BLOCKS_BEHIND) {
+                    self.readiness_indicator.set_unready();
+                } else {
+                    self.readiness_indicator.set_ready();
+                }
+
+                if end <= next_block_index {
+                    break;
+                }
+
+                if self.stop_requested.load(Ordering::SeqCst) {
+                    break;
+                }
```

Comment:
I feel like there should be a simpler way to do this logic, but it's escaping me at this time.